### PR TITLE
Add more hook languages into dredd init

### PR DESCRIPTION
**Pull request by abtris**
By http://dredd.readthedocs.org/en/latest/ are PHP, Go available in Dredd but missing in `dredd init`

### CODE REVIEW

**Comment by honzajavorek** [comment]
Thanks! There is an issue for this: #321 It mentions we should also add it to the `--language` switch. Could you please look into that?

**Comment by abtris** [comment]
ok, I look into.

**Comment by abtris** [comment]
I'm not sure how add [go lang](http://dredd.readthedocs.org/en/latest/hooks-go/) into `hooks-worker-client`.

**Comment by abtris** [comment]
@honzajavorek I think this looks ready. Please make review.

**Comment by honzajavorek** [comment]
Sorry for delay. Could you please also add Perl :camel: from #364? Thanks! :heart_eyes:

**Comment by abtris** [comment]
@honzajavorek I add perl, but I'm not sure if this works.

**Comment by honzajavorek** [comment]
It should, presence of the `dredd-hooks-{language}` command is tested also by the Cucumber test suite. Also see https://github.com/ungrim97/Dredd-Hooks/blob/master/script/dredd-hooks-perl.

File: src/dredd-command.coffee [hunk 1]
```diff
@@ -121,6 +121,12 @@ class DreddCommand
           console.log "  $ gem install dredd_hooks"
         else if config['language'] == 'python'
           console.log "  $ pip install dredd_hooks"
+        else if config['language'] == 'php'
+          console.log "  $ composer require ddelnano/dredd-hooks-php:~1.0.0 --dev"
+        else if config['language'] == 'perl'
+          console.log "  $ cpanm Dredd::Hooks"
+        else if config['language'] == 'go'
+          console.log "  $ go get github.com/snikch/goodman"
```
Comment (honzajavorek) [inline src/dredd-command.coffee]: Does this really work? It's not in the README of the project (there are no installation guidelines :( ).

File: src/dredd-command.coffee [hunk 2]
```diff
@@ -121,6 +121,12 @@ class DreddCommand
           console.log "  $ gem install dredd_hooks"
         else if config['language'] == 'python'
           console.log "  $ pip install dredd_hooks"
+        else if config['language'] == 'php'
+          console.log "  $ composer require ddelnano/dredd-hooks-php:~1.0.0 --dev"
```
Comment (honzajavorek) [inline src/dredd-command.coffee]: Why we do specify version here? And what does the `--dev` mean?

**Comment by netmilk** [comment]
@abtris @honzajavorek Guys! Beware! I suspect there's a typo in the original `doc/hooks-perl.md` documentation for the Perl hooks. It should be `cpan` no `cpanm`. 

@w-vi, would you mind confirming?

@honzajavorek If it is a typo, would you mind fixing it in the doc?

**Comment by abtris** [comment]
@netmilk both exists `cpan` and `cpanm`. I check it. But I'm not sure what is right.

File: test/unit/hooks-worker-client-test.coffee [hunk 3]
```diff
@@ -176,6 +176,60 @@ describe 'Hooks worker client', () ->
...
+              logs.push args.join(" ")
+
+            consoleStub.log.restore()
+            whichStub.which.restore()
+
+            assert.include logs.join(", "), "composer require ddelnano/dredd-hooks-php"
+            done()
+          , 2200
+
+
```
Comment (honzajavorek) [inline test/unit/hooks-worker-client-test.coffee]: Missing test for Perl.

File: src/dredd-command.coffee (hunk 2 above)
Comment (abtris) [inline src/dredd-command.coffee]: Similar `devDependecies` in nodejs.

File: test/unit/dredd-command-test.coffee [hunk 4]
```diff
@@ -272,6 +272,61 @@ describe "DreddCommand class", () ->
       it 'should save configuration', ->
         assert.isTrue configUtilsStub.save.called
 
+    describe '"init"', ->
```
Comment (honzajavorek) [inline test/unit/dredd-command-test.coffee]: Aren't these tests redundant? I'm not sure how important the language is for this particular test. I think it tests just basic functionality of the `dredd init` command itself.

File: src/dredd-command.coffee (hunk 2 above)
Comment (honzajavorek) [inline src/dredd-command.coffee]: Ah, thanks!

File: test/unit/dredd-command-test.coffee (hunk 4 above)
Comment (abtris) [inline test/unit/dredd-command-test.coffee]: Yes, but helping with coverage for lines with others options (php, python, perl).

**Comment by honzajavorek** [comment]
I can fix `cpan` / `cpanm` in #368 if there are mistakes. Maybe @ungrim97 himself can give us a hint?

File: test/unit/dredd-command-test.coffee (hunk 4 above)
Comment (honzajavorek) [inline test/unit/dredd-command-test.coffee]: I don't know what's best practice in that case :( Having several tests just to keep coverage sounds like smell to me, but YMMV. Calling @netmilk for an opinion.

If we are keeping them, can we at least rename [this](https://github.com/apiaryio/dredd/pull/362/files#diff-9583335aa63e76771edff5e02cf4a0c2R257) to

```diff
- describe '"init"', ->
+ describe '"init" (nodejs)', ->
```

and then add the language also to the new ones?


```diff
- describe '"init"', ->
+ describe '"init" (python)', ->
```

etc.

**Comment by ungrim97** [comment]
cpanm is the preferred installation manager for perl (https://metacpan.org/pod/App::cpanminus). If you are using the perl pre installed on the system rather than installing it via perlbrew/plenv then it is likely not already installed. 

cpan is the client that comes packaged with system builds of perl and should work fine (cpan Dredd::Hooks) but may require sudo access if its installing against system perl.

travis (and most perl devs) install perl using either [perlbrew](http://perlbrew.pl/) or [plenv](https://github.com/tokuhirom/plenv) and cpanm to install modules to.

**Comment by honzajavorek** [comment]
@ungrim97 Thanks for clarification! :camel: I'd go with `cpanm` then the same way we rely on people being able to get `pip` on their machine, regardless what they're provided with by default.

**Comment by w-vi** [comment]
`cpanm` FTW http://www.cpan.org/modules/INSTALL.html

**Comment by honzajavorek** [comment]
Thanks! :balloon:

**Pull request merged**